pkg/apis/monitoring/v1alpha1/types.go

```diff
@@ -269,3 +272,25 @@ type ThanosQuerierSpec struct {
 // It should always be reconstructable from the state of the cluster and/or outside world.
 type ThanosQuerierStatus struct {
 }
+
+// SecretKeySelector selects a key of a secret
+type SecretKeySelector struct {
+	// The name of the secret in the object's namespace to select from.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+	// The key of the secret to select from.  Must be a valid secret key.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:Required
+	Key string `json:"key"`
+}
+
+// WebTLSConfig contains configuration to enable TLS on web endpoints
+type WebTLSConfig struct {
+	// Secret containing the TLS key for the server
+	Key SecretKeySelector `json:"key"`
+	// Secret containing the TLS certificate for the server
+	Cert SecretKeySelector `json:"cert"`
+	// Secret containing the CA certificate to verify the server certificates
+	CA SecretKeySelector `json:"ca"`
```

Comment:
(nit) more verbosity is better here IMHO.
Also do we make it required?
```suggestion
	// +kubebuilder:validation:Required
	CertificateAuthority SecretKeySelector `json:"certificateAuthority"`
```

Reply:
Pretty sure required is the default, but sure lets be explicit just in case. Thank you for all the other suggestions. They make sense, I'll include them in the PR.